app/scripts/app.js

```diff
@@ -1,170 +1,182 @@
+  // ... twice the health percentage of any given object
+  // hex is just a color, can be hexadecimal or a string
+  // anyPlayerOrEnemy is an object of the Player() class, will add more soon
+  function Healthbar(width, height, hex, anyPlayerOrEnemy) {
+    this.width = width;
+    this.height = height;
+    this.color = hex;
+    this.canvasPosition = {
+      x: anyPlayerOrEnemy.pos.x,
+      y: anyPlayerOrEnemy.pos.y - 20,
+    };
+  }
 
-    if (delta > interval) {
-      then = now - (delta % interval);
+  // all health bars will need their own draw and update functions
+  // here they are in the prototype
+  Healthbar.prototype = {
+    drawBar: function (anyPlayerOrEnemy) {
+        if (anyPlayerOrEnemy.health.percent) {
+          var num = anyPlayerOrEnemy.health.percent;
+          csv.fillStyle = this.color;
+          csv.fillRect(this.canvasPosition.x, this.canvasPosition.y, this.width, this.height);
+        } else {
+          console.log('In order to determine the length of this health bar, please input a player' +
+            ' or enemy object with a valid .health.percent property.');
+          return;
+        }
+      },
+
+    update: function (anyPlayerOrEnemy) {
+      this.canvasPosition.x = anyPlayerOrEnemy.pos.x;
+      this.canvasPosition.y = anyPlayerOrEnemy.pos.y - 20;
+      this.width = anyPlayerOrEnemy.health.percent * 2;
+    },
+  };
 
-      player.update();
-      $GAME.drawBoard(csv);
-      $GAME.drawPlayer(csv, player);
-      player = $GAME.nextFrame(player);
-    }
+  var playerHealth = new Healthbar(200, 10, 'green', player);
+
+  function updateEverythingThenDraw(timeBetweenFrames) {
+    player.update(timeBetweenFrames);
+    player = $GAME.nextFrame(player);
+    $GAME.drawBoard(csv);
+    $GAME.drawPlayer(csv, player);
+    player.health.percent -= .1;
+    playerHealth.update(player);
+    playerHealth.drawBar(player);
   }
 
-  function startGame() {
-    // set the canvas to the full page size
-    c.width = screenWidth;
-    c.height = screenHeight;
-
-    player = new Player();
-
-    player.frame.frames = 9;
-    player.frame.width = (500/9); // 55.555555555...6
-    player.frame.height = (519/9); // 57.6666666...7
-    player.pos.x = screenWidth/2 - ((500/9)/2);
-    player.pos.y = screenHeight/2 - ((519/9)/2);
-
-    /**
-     * the avatarSrc property is later assigned to player.avatar.src
-     * (src is a property of an HTMLImageElement)
-     * dimensions: 500x519
-     *
-     * @type {String}
-     */
-    player.avatarSrc = 'images/players/001.png';
-
-    // start the animation loop
-    window.requestAnimationFrame(tick);
-
-    window.addEventListener('keydown', keyDown, false);
-	
-    window.addEventListener('keyup', keyUp, false);
+  function startGame(fps) {
+    fpsInterval = 1000 / fps;
+    then = Date.now();
+    startTime = then;
+    tick();
   }
 
-  startGame();
+  function tick() {
+
+    if (player.health.percent < 0) {
```

Comment:
Yeah I thought about that while I was laying down and that would be better I just changed it to less than because for some reason the value was never exactly 0, more like .000000001 or something similar